source/adios2/core/IO.cpp

```diff
@@ -45,6 +45,8 @@ void IO::SetIOMode(const IOMode ioMode) { m_IOMode = ioMode; };
 
 void IO::SetParameters(const Params &parameters) { m_Parameters = parameters; }
 
+const Params &IO::GetParameters() { return m_Parameters; }
```

Comment:
the const will trigger a warning on some compilers: it's meaningless at the beginning of a function